workspace/src/main/kotlin/com/cosmotech/workspace/azure/WorkspaceServiceImpl.kt

```diff
@@ -372,6 +373,22 @@ internal class WorkspaceServiceImpl(
     return WorkspaceAccessControl(rbacAccessControl.id, rbacAccessControl.role)
   }
 
+  override fun updateWorkspaceAccessControl(
+      organizationId: String,
+      workspaceId: String,
+      identityId: String,
+      workspaceRole: WorkspaceRole
+  ): WorkspaceAccessControl {
+    val workspace = findWorkspaceByIdNoSecurity(organizationId, workspaceId)
+    csmRbac.verify(workspace.getRbac(), PERMISSION_WRITE_SECURITY)
+    csmRbac.getAccessControl(workspace.getRbac(), identityId)
+    val rbacSecurity = csmRbac.setUserRole(workspace.getRbac(), identityId, workspaceRole.role)
+    workspace.setRbac(rbacSecurity)
+    cosmosTemplate.upsertAndReturnEntity("${organizationId}_workspaces", workspace)
+    val rbacAccessControl = csmRbac.getAccessControl(workspace.getRbac(), identityId)
+    return WorkspaceAccessControl(rbacAccessControl.id, rbacAccessControl.role)
```

Comment:
You can stored and used the returned value from  `upsertAndReturnEntity` here.

something like this :
```
    val updatedWorkspace = cosmosTemplate.upsertAndReturnEntity(coreOrganizationContainer, organization)
    val rbacAccessControl = csmRbac.getAccessControl(updatedWorkspace.getRbac(), identityId)
    return WorkspaceAccessControl(rbacAccessControl.id, rbacAccessControl.role)
```